### ISSUE TITLE
Docs: [prefer-nullish-coalescing] streamline code examples 

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10566]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The code examples 
1. don't use our standard tabs approach to displaying incorrect vs correct code
2. Repeat the same snippet over and over unnecessarily (the intention was probably to show how the fixer works, but it just isn't great how it's done.)
   ```ts
   const foo: any = 'bar';
   foo ?? 'a string';
   foo ?? 'a string';
   foo ?? 'a string';
   foo ?? 'a string';
   
   const foo: string | undefined = 'bar';
   foo ?? 'a string';
   foo ?? 'a string';
   
   const foo: string | null = 'bar';
   foo ?? 'a string';
   foo ?? 'a string';
   ```
   
Let's use the tabs and streamline the examples a bit. Also let's remove the duplicate identifiers so that the code is valid TS as written.

### Affected URL(s)

https://typescript-eslint.io/rules/prefer-nullish-coalescing/

**Comment by JoshuaKGoldberg** [comment]
Agreed. cc @OlivierZal from #10517 - I'm ok with waiting until after #10517, so this isn't blocked on reviewing the functional & test changes there.

**Comment by OlivierZal** [comment]
Hi, can this issue be unblocked given that the related PR has been merged and that https://github.com/typescript-eslint/typescript-eslint/pull/10708 is ready for review? This PR doesn’t affect the documentation so there shouldn’t be any conflict but I’ll rebase if needed when it is merged.

**Comment by JoshuaKGoldberg** [comment]
Yes, good call!
